cmd/paramgen/main.go

```diff
@@ -0,0 +1,152 @@
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"flag"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/conduitio/conduit-connector-sdk/cmd/paramgen/internal"
+)
+
+type Args struct {
+	output     string
+	structName string
+	path       string
+}
+
+func main() {
+	log.SetFlags(0)
+	log.SetPrefix("paramgen: ")
+
+	// parse the command arguments
+	args := parseArgs()
+
+	// parse the sdk parameters
+	params, err := internal.ParseParameters(args.path, args.structName)
+	if err != nil {
+		log.Fatalf("paramgen failed to parse parameters: %v\n", err)
+	}
+
+	// convert the parameter map to a string
+	str := paramsToString(params)
+
+	processTemplate(str, args)
+}
+
+const (
+	outputDirName = "/paramgen"
+	paramGenTmpl  = `// Code generated by ParamGen. DO NOT EDIT.
+// Source: github.com/conduitio/conduit-connector-sdk/cmd/paramgen
+
+// Package paramgen is a generated ParamGen package.
+package paramgen
+
+import sdk "github.com/conduitio/conduit-connector-sdk"
+
+func getParameters() map[string]sdk.Parameter {
+  return {{.}}
+}
+`
+)
+
+// processTemplate creates and executes the paramgen template, then writes the output file under the directory ./paramgen.
+func processTemplate(str string, args Args) {
+	// create the go template
+	t := template.Must(template.New("").Parse(paramGenTmpl))
+
+	var processed bytes.Buffer
+	// execute the template
+	err := t.Execute(&processed, str)
+	if err != nil {
+		log.Fatalf("error executing template: %v\n", err)
+	}
+
+	// format the output as Go code in the “gofmt” style
+	formatted, err := format.Source(processed.Bytes())
+	if err != nil {
+		log.Fatalf("Could not format processed template: %v\n", err)
+	}
+
+	// create output directory if it does not exist
+	outputDir := strings.TrimSuffix(args.path, "/") + outputDirName
+	_, err = os.Stat(outputDir)
+	if os.IsNotExist(err) {
+		err = os.Mkdir(outputDir, os.ModePerm)
+		if err != nil {
+			log.Fatalf("could not create directory: %v\n", err)
+		}
+	}
+
+	// create the output file and write data
+	outputPath := outputDir + "/" + args.output
+	f, err := os.Create(outputPath)
+	if err != nil {
+		log.Fatalf("could not create file: %v\n", err)
+	}
+	w := bufio.NewWriter(f)
+	_, err = w.WriteString(string(formatted))
+	if err != nil {
+		log.Fatalf("error writing to a file: %v\n", err)
+	}
+	w.Flush()
+	log.Printf("output file created: %s\n", outputPath)
+}
+
+func paramsToString(params map[string]sdk.Parameter) string {
+	str := fmt.Sprintf("%#v\n", params)
+	str = strings.ReplaceAll(str, "{", "{\n")
+	str = strings.ReplaceAll(str, "}}", "},\n}")
+	return str
+}
+
+func parseArgs() Args {
+	flags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	var (
+		output     = flags.String("output", "conduit-config.go", "name of the output file")
+		structName = flags.String("struct", "Config", "the configuration struct name")
```

Comment:
I wouldn't have a default for this option. In fact, I would change this to be an argument instead of a flag and make it required for the user to provide it.